hecuba_core/src/py_interface/HCache.cpp

```diff
@@ -63,9 +63,16 @@ static PyObject *put_row(HCache *self, PyObject *args) {
             return NULL;
         }
     }
+    std::set<std::string> keys;
+    PyObject *elem = NULL, *format_text = NULL;
+    for (uint16_t value_i = 0; value_i < PyList_Size(py_union_k); ++value_i) {
+        format_text = PyUnicode_AsUTF8String(PyList_GetItem(py_union_k, value_i));
```

Comment:
This method access the internal data of the String, but the data may change format from a version of python to another. It would be better if you transform the object to string like we did here (this should work always):

https://github.com/bsc-dd/hecuba/blob/75149b19c560244985ce7095c2afb86d88fbf971/hecuba_core/src/py_interface/UnitParser.cpp#L191